plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/TestingOpenLdapServer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.password.ldap;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Closer;
+import io.trino.testing.TestingProperties;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+
+import javax.naming.Context;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+
+import static io.trino.plugin.password.jndi.JndiUtils.createDirContext;
+import static io.trino.plugin.password.ldap.LdapUtil.MEMBER_OF;
+import static io.trino.plugin.password.ldap.LdapUtil.addAttributesToExistingLdapObjects;
+import static io.trino.plugin.password.ldap.LdapUtil.addLdapDefinition;
+import static io.trino.plugin.password.ldap.LdapUtil.buildLdapGroupObject;
+import static io.trino.plugin.password.ldap.LdapUtil.buildLdapOrganizationObject;
+import static io.trino.plugin.password.ldap.LdapUtil.buildLdapUserObject;
+import static java.lang.String.format;
+
+public class TestingOpenLdapServer
+        implements Closeable
+{
+    private static final String BASE_DISTINGUISED_NAME = "dc=trino,dc=testldap,dc=com";
+
+    private static final int LDAP_PORT = 389;
+
+    private final Closer closer = Closer.create();
+    private final GenericContainer<?> openLdapServer;
+
+    public TestingOpenLdapServer()
+    {
+        openLdapServer = new GenericContainer<>("ghcr.io/trinodb/testing/centos7-oj11-openldap:" + TestingProperties.getDockerImagesVersion())
+                .withExposedPorts(LDAP_PORT)
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy());
```

Comment:
How do we know that LDAP is up and running? Having port available might not be enough.

Reply:
I guess we use the same strategy in PTs too. I don't thing we face any issue there.

Reply:
I think only successful LDAP search gives positive answer. But maybe image internally has some check in init scripts.